kernel/src/hil/digest.rs

```diff
@@ -0,0 +1,38 @@
+//! Interface for Digest
+
+use crate::returncode::ReturnCode;
+
+/// Implement this trait and use `set_client()` in order to receive callbacks.
+pub trait Client<'a> {
+    /// On error `result` will contain device specific error codes
+    fn hash_done(&'a self, result: Result<(), u32>);
+}
+
+pub trait Digest<'a> {
```

Comment:
What does this do? As in what would a user expect to happen if they used a `Digest` interface?

Reply:
It computes a digest (a cryptographic hash) over data.

Reply:
A user shouldn't use the digest interface, the idea is that a `Hmac` virtual device is implemented on top of a `Digest`. This would also allow a `SHA` virtual device on top of the `Digest` and the same hardware can be supported for both (if the HW supports both).